projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -188,24 +200,36 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
 
     const transformedValue: FormInterface = this.transformToFormGroup(obj);
 
-    // for now we throw an error if the transformed value isn't an object with all the expect values
-    // there's an issue to track support for an array https://github.com/cloudnc/ngx-sub-form/issues/9
-    if (Array.isArray(transformedValue)) {
-      throw new ArrayNotTransformedBeforeWriteValueError();
-    }
-
     const missingKeys: (keyof FormInterface)[] = this.getMissingKeys(transformedValue);
     if (missingKeys.length > 0) {
       throw new MissingFormControlsError(missingKeys as string[]);
     }
 
+    this.handleFormArrayControls(transformedValue);
+
     this.formGroup.setValue(transformedValue, {
       emitEvent: false,
     });
     this.formGroup.markAsPristine();
     this.formGroup.markAsUntouched();
   }
 
+  private handleFormArrayControls(obj: any) {
+    Object.entries(obj).forEach(([key, value]) => {
+      if (this.formGroup.get(key) instanceof FormArray && Array.isArray(value)) {
```

Comment:
Note that if we've got an array but the current `AbstractControl` is not a `FormArray` then we do nothing.

This allows us to either handle an array in the current view or just create a `FormControl` that'll have the array as value and pass it to a sub component that will be in charge of that array himself.